addons/mail/models/mail_channel.py

```diff
@@ -180,46 +211,54 @@ def _onchange_moderation(self):
 
     @api.model_create_multi
     def create(self, vals_list):
-        defaults = self.default_get(['image_128'])
-        current_partner = self.env.user.partner_id.id
+        defaults = self.default_get(['image_128', 'public'])
 
-        visibilities = []
+        access_types = []
         for vals in vals_list:
+            # find partners to add from partner_ids
+            partner_ids_cmd = vals.get('channel_partner_ids') or []
+            if any(cmd[0] == 0 for cmd in partner_ids_cmd):
+                raise ValidationError(_('Invalid value when creating a channel with members, only 4 or 6 are allowed.'))
+            partner_ids = [cmd[1] for cmd in partner_ids_cmd if cmd[0] == 4]
+            partner_ids += [cmd[2] for cmd in partner_ids_cmd if cmd[0] == 6]
+
+            # find partners to add from channel_last_seen_partner_ids
+            membership_ids_cmd = vals.get('channel_last_seen_partner_ids') or []
+            if any(cmd[0] == 0 for cmd in partner_ids_cmd):
+                raise ValidationError(_('Invalid value when creating a channel with memberships, only 0 is allowed.'))
+            membership_pids = [cmd[2]['partner_id'] for cmd in membership_ids_cmd if cmd[0] == 0]
+
+            # always add current user to new channel to have right values for
+            # is_pinned + ensure he has rights to see channel
+            partner_ids_to_add = list(set(partner_ids + [self.env.user.partner_id.id]))
+            vals['channel_last_seen_partner_ids'] = membership_ids_cmd + [
+                (0, 0, {'partner_id': pid})
+                for pid in partner_ids_to_add if pid not in membership_pids
```

Comment:
I'm not sure how you could add memberships to a channel you're currently creating.
Since you would have to provide its ID in the (0, 0, {}) command, and it's not yet created.

It would also allow you to add memberships for other channels that the one you're creating, which seems very strange?

Also, there is no "write" support of all this? Only possible at create level?

All in all, this makes my head hurt a bit ^^

Reply:
When using these commands the ORM should add the relation column1 field (`channel_id`) automatically in that case I believe.
This is updating `vals` for one specific channel so it will not be added to other channels.

Reply:
Ok I see it now, "channel_last_seen_partner_ids" is a relationship to the decorated join table and "channel_partner_ids" are the partners from that relationship.

Strange naming + "stored" comment in commit message had me confused.

Thanks for clarifying.